mne/io/bti/bti.py

```diff
@@ -1143,7 +1143,8 @@ def by_name(x):
         data = _read_data(bti_info)
         assert len(data) == len(info['ch_names'])
         self._projector_hashes = [None]
-        super(RawBTi, self).__init__(info, data, verbose=verbose)
+        super(RawBTi, self).__init__(
+            info, data, filenames=[pdf_fname], verbose=verbose)
```

Comment:
Yeah, it would work. Right now BTI doesn't support non-preloaded data, so it's a moot point currently. But in the future when someone wants to do it, they can put whatever they want in `_raw_extras`, one per loaded raw file. That could include whatever params and files need to be parsed in order to construct data in `_read_segment_file`. Actually the EDF reader already does it by storing things like `annot`, see below where this happens:
```
annot = self._raw_extras[fi]['annot']
```
It pulls the annot param from the given file number.